examples/minipc/ChassisTest.cc

```diff
@@ -0,0 +1,193 @@
+*  This program is distributed in the hope that it will be useful,         *
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of          *
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
+*  GNU General Public License for more details.                            *
+*                                                                          *
+*  You should have received a copy of the GNU General Public License       *
+*  along with this program. If not, see <http://www.gnu.org/licenses/>.    *
+*                                                                          *
+****************************************************************************/
+
+#include "main.h"
+#include <memory>
+#include "bsp_print.h"
+#include "bsp_uart.h"
+#include "cmsis_os.h"
+#include "minipc_protocol.h"
+#include "rgb.h"
+#include "chassis.h"
+#include "dbus.h"
+#include "tim.h"
+
+/* 
+  This example receives vx, vy, and vw commands from the Jetson and commands 
+  the chassis. 
+
+  On the Jetson side, the following scripts in robot_driver work with this
+  example: chassis_test.py, cmd_vel_adjuster.py, and keyboard_teleop.py.
+
+  (Note that if you are using cmd_vel_adjuster.py and keyboard_teleop.py, you
+  will need to run robot.py as well. See each script for more information.)
+*/
+
+#define RX_SIGNAL (1 << 0)
+#define DATA_READY_SIGNAL (1 << 1)
+extern osThreadId_t defaultTaskHandle;
+
+static display::RGB* led = nullptr;
+bsp::CAN* can = nullptr;
+control::MotorCANBase* fl_motor = nullptr;
+control::MotorCANBase* fr_motor = nullptr;
+control::MotorCANBase* bl_motor = nullptr;
+control::MotorCANBase* br_motor = nullptr;
+
+control::Chassis* chassis = nullptr;
+remote::DBUS* dbus = nullptr;
+
+osEventFlagsId_t chassis_flag_id;
+
+static float vx = 0;
+static float vy = 0;
+static float vw = 0;
+
+class CustomUART : public bsp::UART {
+ public:
+  using bsp::UART::UART;
+
+ protected:
+  /* notify application when rx data is pending read */
+  void RxCompleteCallback() override final { osThreadFlagsSet(defaultTaskHandle, RX_SIGNAL); }
+};
+
+const osThreadAttr_t chassisTaskAttribute = {.name = "chassisTask",
+                                             .attr_bits = osThreadDetached,
+                                             .cb_mem = nullptr,
+                                             .cb_size = 0,
+                                             .stack_mem = nullptr,
+                                             .stack_size = 256 * 4,
+                                             .priority = (osPriority_t)osPriorityNormal,
+                                             .tz_module = 0,
+                                             .reserved = 0};
+osThreadId_t chassisTaskHandle;
+
+void chassisTask(void* argument) {
+  UNUSED(argument);
+  uint32_t flags;
+  control::MotorCANBase* motors[] = {fl_motor, fr_motor, bl_motor, br_motor};
+
+  float vx_keyboard = 0, vy_keyboard = 0, wz_keyboard = 0;
```

Comment:
nit: Why different treatment for vX and vX_keyboard? Consider using newlines.

Reply:
Fixed.